src/main/io/displayport_hott.c

```diff
@@ -0,0 +1,187 @@
+/*
+ * This file is part of Cleanflight.
+ *
+ * Cleanflight is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Cleanflight is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Cleanflight.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <stdbool.h>
+#include <stdint.h>
+#include <string.h>
+
+#include "platform.h"
+#if defined (USE_HOTT_TEXTMODE) && defined (USE_CMS)
+
+// Fixme:
+// Let the CMS believe we have a bigger display to avoid empty rows and columns
+// Better make this configurable in CMS
+#define ROW_CORRECTION 1
+#define COL_CORRECTION 2
+
+#include "common/utils.h"
+#include "cms/cms.h"
+#include "telemetry/hott.h"
+
+displayPort_t hottDisplayPort;
+
+static int hottDrawScreen(displayPort_t *displayPort)
+{
+    UNUSED(displayPort);
+    return 0;
+}
+
+static int hottScreenSize(const displayPort_t *displayPort)
+{
+    return displayPort->rows * displayPort->cols;
+}
+
+static int hottWriteChar(displayPort_t *displayPort, uint8_t col, uint8_t row, uint8_t c)
+{
+    UNUSED(displayPort);
+
+
+    // Correct fake display dimensions
+    if (row <= ROW_CORRECTION - 1)
+        return 0;
+    else
+        row -= ROW_CORRECTION;
+
+    if (col <= COL_CORRECTION - 1)
+        return 0;
+    else
+        col -= COL_CORRECTION;
+
+    if (row >= HOTT_TEXTMODE_DISPLAY_ROWS || col >= HOTT_TEXTMODE_DISPLAY_COLUMNS)
+        return 0;
+
+    hottTextmodeWriteChar(col, row, c);
+    return 0;
+}
+
+static int hottWriteString(displayPort_t *displayPort, uint8_t col, uint8_t row, const char *s)
+{
+    while (*s) {
+        hottWriteChar(displayPort,  col++, row, *(s++));
+    }
+    return 0;
+}
+
+static int hottClearScreen(displayPort_t *displayPort)
+{
+    for (int row = 0; row < displayPort->rows; row++) {
+        for (int col= 0; col < displayPort->cols; col++) {
+            hottWriteChar(displayPort, col, row, ' ');
```

Comment:
(this is clearing invisible row/cols too)

Reply:
No, hottWriteChar(...) is on displayport_hott.c and checks for invisible row/cols. 
You mean hottTextmodeWriteChar(...) in hott.c

Reply:
`displayPort->rows`/`cols` is larger than actual `hottTextmodeWriteChar` accepted area, so some writes are skipped ... it is only minor detail ...